### ISSUE TITLE
RC1: Unable to connect, please ensure WebSockets are available. A VPN or proxy may be blocking the connection.

### ISSUE THREAD

**Issue by oiBio** [dotnet/aspnetcore#36596]
### Describe the bug
After upgrading to net6 and RC1 my app is unable to initiate a websocket connection for my blazor server app.
Websockets was working fine with net5. Unfortunatly i cant reenable long polling for testing purposes, since the provided work around does not work: see https://github.com/dotnet/docs/issues/26158

### To Reproduce
Unable to reproduce this outside of my application (which is a large blazor app growing since 2019)
Blazor works with web sockets on newly created templates.

### Exceptions (if any)
```
blazor.server.js:1 [2021-09-16T08:23:55.484Z] Information: Normalizing '_blazor' to 'https://localhost:5001/_blazor'.
blazor.server.js:1 WebSocket connection to 'wss://localhost:5001/_blazor?id=r5ooMp8894ByZYsYc-fOTA' failed: 
(anonymous) @ blazor.server.js:1
blazor.server.js:1 [2021-09-16T08:23:55.558Z] Information: (WebSockets transport) There was an error with the transport.
blazor.server.js:1 [2021-09-16T08:23:55.559Z] Error: Failed to start the transport 'WebSockets': Error: WebSocket failed to connect. The connection could not be found on the server, either the endpoint may not be a SignalR endpoint, the connection ID is not present on the server, or there is a proxy blocking WebSockets. If you have multiple servers check that sticky sessions are enabled.
log @ blazor.server.js:1
blazor.server.js:1 [2021-09-16T08:23:55.559Z] Error: Failed to start the connection: Error: Unable to connect to the server with any of the available transports. Error: WebSockets failed: Error: WebSocket failed to connect. The connection could not be found on the server, either the endpoint may not be a SignalR endpoint, the connection ID is not present on the server, or there is a proxy blocking WebSockets. If you have multiple servers check that sticky sessions are enabled.
log @ blazor.server.js:1
blazor.server.js:1 [2021-09-16T08:23:55.559Z] Error: Error: Unable to connect to the server with any of the available transports. Error: WebSockets failed: Error: WebSocket failed to connect. The connection could not be found on the server, either the endpoint may not be a SignalR endpoint, the connection ID is not present on the server, or there is a proxy blocking WebSockets. If you have multiple servers check that sticky sessions are enabled.
log @ blazor.server.js:1
blazor.server.js:1 [2021-09-16T08:23:55.559Z] Error: Failed to start the circuit.
log @ blazor.server.js:1
```

### Further technical details
- ASP.NET Core version | dotnet 6 RC1
- Include the output of `dotnet --info`
```
.NET SDK (gemäß "global.json"):
 Version:   6.0.100-rc.1.21458.32
 Commit:    d7c22323c4

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.100-rc.1.21458.32\

Host (useful for support):
  Version: 6.0.0-rc.1.21451.13
  Commit:  d7619cd4b1

.NET SDKs installed:
  2.2.207 [C:\Program Files\dotnet\sdk]
  3.1.200-preview-014977 [C:\Program Files\dotnet\sdk]
  3.1.200-preview-015002 [C:\Program Files\dotnet\sdk]
  3.1.300-preview-015048 [C:\Program Files\dotnet\sdk]
  3.1.300-preview-015095 [C:\Program Files\dotnet\sdk]
  3.1.302 [C:\Program Files\dotnet\sdk]
  3.1.400-preview-015151 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.2.20479.15 [C:\Program Files\dotnet\sdk]
  5.0.100 [C:\Program Files\dotnet\sdk]
  5.0.201 [C:\Program Files\dotnet\sdk]
  5.0.301 [C:\Program Files\dotnet\sdk]
  5.0.302 [C:\Program Files\dotnet\sdk]
  5.0.401 [C:\Program Files\dotnet\sdk]
  6.0.100-preview.4.21255.9 [C:\Program Files\dotnet\sdk]
  6.0.100-rc.1.21458.32 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-preview.6.21355.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-rc.1.21452.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-preview.6.21352.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-rc.1.21451.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.19 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.0-rc.1.21451.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version
-  VS 2022 Preview 4

**Comment by javiercn** [comment]
@oiBio thanks for contacting us.

We've disabled transports other than websockets by default on Blazor server as of RC1. This likely means (as the error says) that something on your environment is blocking websocket connections. We strongly recommend you use websockets in conjunction with Blazor Server for best performance. 

However if for some reason that's not possible I believe there is a way to change it. @TanayParikh can you add details here?

/cc: @danroth27 FYI

**Comment by oiBio** [comment]
when i reverse the changes to net5 i get wss connections all along. the enviroment did not change
tried all different browsers and still get this message :-/

**Comment by javiercn** [comment]
@oiBio thanks for the additional details.

We'll take a look at it. Can you give us more details about your environment? (Do you have a corporate proxy, firewall, VPN, or similar). Are you deploying to a well known environment (like Azure websites or similar)

**Comment by oiBio** [comment]
nothing in that manner. no proxy, no vpn and only the default windows firewall - which was never a problem.
We normally deploy to a iis enviroment and use the default blazor signal r connection.

**Comment by TanayParikh** [comment]
> However if for some reason that's not possible I believe there is a way to change it. @TanayParikh can you add details here?

The breaking changes announcement with details on how to re-enable long polling can be found here:

https://docs.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/blazor-long-polling-fallback

**Comment by TanayParikh** [comment]
> Unfortunatly i cant reenable long polling for testing purposes, since the provided work around does not work: see dotnet/docs#26158

Looks up be missing the auto start attribute and thus blazor is starting before it can be configured. I left a comment in the docs issue.

**Comment by TanayParikh** [comment]
> when i reverse the changes to net5 i get wss connections all along. the enviroment did not change
> tried all different browsers and still get this message :-/

Would you mind pasting the console log output from when you revert to .NET 5 and web sockets are working? Want to ensure it didn't silently fallback to long polling in that case.

**Comment by danroth27** [comment]
Note that you typically need to configure IIS to enable WebSockets: https://docs.microsoft.com/iis/get-started/whats-new-in-iis-8/iis-80-websocket-protocol-support#step-by-step-instructions.

**Comment by oiBio** [comment]
TLDR: It was a expired ssl certificate

Hello all,

thank you for the active participation and help. I have found the problem now, it was due to an expired localhost certificate that overlapped a bit with my test.

Now that the long polling fallback was implemented correctly and it worked, I checked my net5 version again and was able to find a faulty web socket connection here as well. Thanks to Taney for insisting on this. I was 100% convinced that my console always printed something link "Information: WebSocket connected to wss://localhost:5001/_blazor?id=Q4A6CJUScr57ti_aIG65Rg." or similar. But by double checking the expired certificate and a rejected WSS stood out and so LongPolling was active....

Sorry for the effort. But at least a small gap in the docs was noticed through this.
